raiden/ui/web/src/app/services/raiden.service.ts

```diff
@@ -341,14 +356,19 @@ export class RaidenService {
         tokenAddress: string,
         refresh: boolean = true,
     ): Observable<UserToken | null> {
+
         const tokenContractInstance = this.tokenContract.at(tokenAddress);
         const tokenMap = this.userTokens;
         const userToken: UserToken | null | undefined = tokenMap[tokenAddress];
 
-        const balance$: Observable<number> = bindNodeCallback((addr: string, cb: CallbackFunc) =>
-            tokenContractInstance.balanceOf(addr, this.zoneEncap(cb)),
-        )(this.raidenAddress).pipe(
-            map((balance) => balance.toNumber())
+        const balance$: Observable<number> = this.getRaidenAddress().pipe(
```

Comment:
This will make a new http request to `/address` everytime a `getUserToken` is called (as well as any observable which uses `getRaidenAddress`. Please, make `raidenAddress$` a single, readonly instance, instead of a method, and `shareReplay(1)` it, as the address is not expected to change

Reply:
Yes that would make sense. I am on it.

Reply:
There were 2-3 usages only so I just inlined the `getRaidenAddress` after making it return `raidenAddress$`.